### PR TITLE
chore: Made <details> tags pretty 

### PR DESCRIPTION
**Pull request by dayvster**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Styled the details tags, added an animation script included it in the docs `pageContent.tsx` made sure that it loads in async not to block the page load.

---

## Screenshots

image

image

💯

### CODE REVIEW

File: www/public/animateDetails.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/**
```
Comment (c-ehrlich) [inline www/public/animateDetails.js:1]: The functionality seems good based on the preview deployment, but I don't love the idea of inlining a .js file. That being said I've never implemented something like this. Is this how other projects do it as well?

Comment (dayvster) [reply www/public/animateDetails.js:1]: @c-ehrlich I suppose the JS code could simply be moved directly into `pageContent.astro` and only refactored out when/if it needs to be re-used somewhere else?

Comment (nexxeln) [reply www/public/animateDetails.js:1]: > @c-ehrlich I suppose the JS code could simply be moved directly into `pageContent.astro` and only refactored out when/if it needs to be re-used somewhere else?

Yeah I would like if it was in an astro file, or just a file in `src/utils` or whatever. Putting a js file in public feels weird.

Comment (dayvster) [reply www/public/animateDetails.js]: @c-ehrlich @nexxeln refactored it from `public/animateDetails.js` into `pageContent.astro` should we mark this convo as resolved?

**Comment by nexxeln** [comment]
CI is failing for some reason

**Comment by dayvster** [comment]
> CI is failing for some reason

Noticed, it expects the script in pageContent to be typescript and is complaining about missing types.

**Comment by juliusmarminge** [comment]
> > CI is failing for some reason
> 
> 
> 
> Noticed, it expects the script in pageContent to be typescript and is complaining about missing types. 

You can add the `is:inline` directive or make it properly typed

**Comment by dayvster** [comment]
> > > CI is failing for some reason
> > 
> > 
> > Noticed, it expects the script in pageContent to be typescript and is complaining about missing types.
> 
> You can add the `is:inline` directive or make it properly typed

Done and done.
 
> EDIT: but as @c-ehrlich pointed out, is all this JavaScript needed to make it animate the "fold"?

Sadly I'd say yes, I tried doing it with pure CSS/tailwind at first and it didn't quite work.

File: www/public/animateDetails.js (hunk 1 above)
Comment (dayvster) [reply www/public/animateDetails.js]: @nexxeln @c-ehrlich refactored from `public/animateDetails.js` into `docs/pageContent.astro` as requested.

**Comment by juliusmarminge** [comment]
> > > > CI is failing for some reason
> > > 
> > > 
> > > Noticed, it expects the script in pageContent to be typescript and is complaining about missing types.
> > 
> > 
> > You can add the `is:inline` directive or make it properly typed
> 
> Done and done.
> 
> > EDIT: but as @c-ehrlich pointed out, is all this JavaScript needed to make it animate the "fold"?
> 
> Sadly I'd say yes, I tried doing it with pure CSS/tailwind at first and it didn't quite work.

Alright Lighthouse Performance still 100 so all good :)

Maybe leave a FIXME comment to make that script TypeScript sometime in the future?

File: www/src/components/docs/pageContent.astro [hunk 2]
```diff
@@ -101,3 +101,122 @@ const title = frontmatter.title;
     block.appendChild(copyButton);
   });
 </script>
+<script is:inline>
```
Comment (juliusmarminge) [inline www/src/components/docs/pageContent.astro:104]:
```suggestion
<script is:inline>
// FIXME: We can make this properly typed later
```

File: www/src/components/docs/pageContent.astro [hunk 3]
```diff
@@ -101,3 +101,123 @@ const title = frontmatter.title;
...
+// FIXME: We can make this properly typed later
```
Comment (juliusmarminge) [inline www/src/components/docs/pageContent.astro:105]:
```suggestion
  //FIXME: We can make this properly typed later
```

**Review by juliusmarminge — APPROVED** [review]
Fancy :)

**Review by nexxeln — APPROVED** [review]

**Pull request merged**